esp_repl/esp_repl.c

```diff
@@ -0,0 +1,214 @@
+
+/*
+* SPDX-FileCopyrightText: 2025 Espressif Systems (Shanghai) CO LTD
+*
+* SPDX-License-Identifier: Apache-2.0
+*/
+#include <stdbool.h>
+#include <string.h>
+#include "freertos/FreeRTOS.h"
+#include "freertos/semphr.h"
+#include "esp_repl.h"
+#include "esp_err.h"
+#include "esp_commands.h"
+#include "esp_linenoise.h"
+typedef enum {
+    ESP_REPL_STATE_RUNNING,
+    ESP_REPL_STATE_STOPPED
+} esp_repl_state_e;
+
+typedef struct esp_repl_state {
+    esp_repl_state_e state;
+    SemaphoreHandle_t mux;
+} esp_repl_state_t;
+
+typedef struct esp_repl_instance {
+    esp_repl_handle_t self;
+    esp_repl_config_t config;
+    esp_repl_state_t state;
+} esp_repl_instance_t;
+
+#define ESP_REPL_CHECK_INSTANCE(handle) do {                                                          \
+        if((handle == NULL) || ((esp_repl_instance_t*)handle->self != (esp_repl_instance_t*)handle)) {    \
+            return ESP_ERR_INVALID_ARG;                                                                   \
+        }                                                                                                 \
+    } while(0)
+
+esp_err_t esp_repl_create(const esp_repl_config_t *config, esp_repl_handle_t *out_handle)
+{
+    if (!config || !out_handle) {
+        return ESP_ERR_INVALID_ARG;
+    }
+
+    if ((config->linenoise_handle == NULL) ||
+            (config->max_cmd_line_size == 0)) {
+        return ESP_ERR_INVALID_ARG;
+    }
+
+    esp_repl_instance_t *instance = malloc(sizeof(esp_repl_instance_t));
+    if (!instance) {
+        return ESP_ERR_NO_MEM;
+    }
+
+    instance->config = *config;
```

Comment:
We must check that `linenoise_handle` is not NULL since it's not optional

Reply:
it is already checked at the beginning of the function